File: algorithms/sort/selection/algo.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<int> arr(n);

    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    for (int next = 0; next < n; next++)
    {

        int biggest = next;

        for (int i = next + 1; i < n; i++)
        {
            if (arr[i] < arr[biggest])
            {
                biggest = i;
            }
        }

        swap(arr[biggest], arr[next]);
    }

    for (int i = 0; i < n; i++)
    {
        if (i > 0)
            cout << " ";

        cout << arr[i];
    }
    cout << endl;
    return 0;
}

File: algorithms/sort/selection/algo.rs
use std::io::{self, BufRead};

fn selection_sort(n:u32, &arr: Vec<i32>) => Vec<i32> {

    
}

fn main() {
    let stdin = io::stdin();
    let mut iterator = stdin.lock().lines();
    let n: u32 = iterator.next().unwrap().unwrap().parse().unwrap();
    println!("n is {:#?}", n);
    let input = iterator.next().unwrap().unwrap();
    let arr: Vec<i32> = input.split(' ').into_iter().map(|x| x.parse::<i32>().unwrap()).collect();
    println!("b is {:#?}", &b);

    let sorted_arr = selection_sort(n, arr);

    println!("after sorting: {:#?}", &selection_sort(n, arr))
}